src/Compilers/Core/Portable/Compilation/ITypeDeclaration.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis
+{
+    public interface ITypeDeclaration
+    {
+        string Name { get; }
+
+        TypeKind Kind { get; }
+
+        Accessibility Accessibility { get; }
+
+        int Arity { get; }
+    }
```

Comment:
i would personally prefer the tree just be exposed, instead of just these single type nodes that are flattened into an array.